### PR TITLE
feat: pass rawHeaders

### PR DESCRIPTION
**Pull request by MoLow**
add rawHeaders whenever we return headers.
hope this PR makes sense
closes https://github.com/nodejs/undici/issues/1213

### CODE REVIEW

**Review by ronag — COMMENTED** [review]
I'm concerned that this causes reduced performance for an edge case.

**Comment by ronag** [comment]
Can we make this an option on the request instead? i.e. whether or not to return raw headers. That way it doesn't affect performance and you get exactly what you want in the fastest way possible.

e.g.

```js
    const {
        statusCode,
        headers,
        trailers,
        body
    } = await undici.request('http://url', { headers: 'raw' });
```

**Comment by MoLow** [comment]
> Can we make this an option on the request instead? i.e. whether or not to return raw headers. That way it doesn't affect performance and you get exactly what you want in the fastest way possible.
> 
> e.g.
> 
> ```js
>     const {
>         statusCode,
>         headers, // rawHeaders
>         trailers,
>         body
>     } = await undici.request('http://url', { responeHeaders: 'raw' });
> ```
> 
> I'm not sure what the option should be called but I think something like this.

@ronag  makes sense to me,

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/core/util.js [hunk 1]
```diff
@@ -200,6 +200,10 @@ function parseHeaders (headers, obj = {}) {
   return obj
 }
 
+function parseRawHeaders (headers) {
+  return headers.map(header => header.toString());
+}
```
Comment (ronag) [inline lib/core/util.js:205]: Do we even need this? I think it's fine to leave this conversion to user space, no? That way we don't have to add another option when someone wants the "raw" headers as buffers.

**Pull request merged**